lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionImpl.kt

```diff
@@ -0,0 +1,74 @@
+package dev.sargunv.maplibrecompose.core.expression
+
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+
+internal class ExpressionImpl private constructor(override val value: Any?) :
+  BooleanExpression,
+  FloatExpression,
+  IntExpression,
+  DpExpression,
+  StringExpression,
+  EnumExpression<LayerPropertyEnum>,
+  ColorExpression,
+  MapExpression,
+  VectorExpression, // also provides ListExpression<*> (with type erasure)
+  OffsetExpression,
+  DpOffsetExpression,
+  PaddingExpression,
+  FormattedExpression,
+  ImageExpression,
+  CollatorExpression,
+  Interpolation {
+  internal inline fun <reified T : Expression> cast(): T = this as T
+
+  internal companion object {
+    val ofNull = ExpressionImpl(null)
+    val ofTrue = ExpressionImpl(true)
+    val ofFalse = ExpressionImpl(false)
+
+    private val constSmallInts = Array(512) { ExpressionImpl(it.toFloat()) }
+    private val constSmallFloats = Array(512) { ExpressionImpl(it.toFloat() / 20f) }
+    private val ofEmptyString = ExpressionImpl("")
+    private val ofTransparent = ExpressionImpl(Color.Transparent)
+    private val ofBlack = ExpressionImpl(Color.Black)
+    private val ofWhite = ExpressionImpl(Color.White)
+    private val ofEmptyList = ExpressionImpl(emptyList<Any?>())
+    private val ofZeroOffset = ExpressionImpl(Offset.Zero)
+    private val ofZeroPadding = ExpressionImpl(ZeroPadding)
+
+    private val ofEmptyMap = ExpressionImpl(emptyMap<String, Any?>())
+
+    private fun isSmallInt(f: Float) = f >= 0 && f < 512 && f.toInt().toFloat() == f
+
+    fun ofFloat(float: Float) =
```

Comment:
I am surprised that there isn't a 

```suggestion

    fun ofInt(int: Int) = 
      if (int >= 0 && int < constSmallInts.size) constSmallInts[int]
      else ExpressionImpl(int)
      
    fun ofFloat(float: Float) =
```

Reply:
`const` in the DSL has a variant for int so it didn't seem necessary, though I agree it's weird, can add it